extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/handlers/DevClasspathStaticHandler.java

```diff
@@ -0,0 +1,138 @@
+package io.quarkus.vertx.http.runtime.handlers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.impl.MimeMapping;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * {@link StaticHandler} implementation to handle static resources using the Classpath.
+ * This is meant to be used on {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT} mode.
+ */
+public class DevClasspathStaticHandler implements Handler<RoutingContext> {
+
+    private static final Logger LOG = Logger.getLogger(DevClasspathStaticHandler.class);
+    private static final Set<HttpMethod> ALLOWED_HTTP_METHODS = Set.of(HttpMethod.GET, HttpMethod.HEAD, HttpMethod.OPTIONS);
+    private static final int HTTP_STATUS_OK = 200;
+    private final Set<String> generatedResources;
+    private final Set<String> compressedMediaTypes;
+    private final ClassLoader currentClassLoader;
+    private final boolean enableCompression;
+    private final String indexPage;
+
+    public DevClasspathStaticHandler(Set<String> generatedResources, GeneratedStaticResourcesHandlerOptions options) {
+        this.generatedResources = generatedResources;
+        this.compressedMediaTypes = options.getCompressMediaTypes();
+        this.currentClassLoader = Thread.currentThread().getContextClassLoader();
+        this.enableCompression = options.isEnableCompression();
+        this.indexPage = options.getIndexPage();
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        if (!ALLOWED_HTTP_METHODS.contains(context.request().method())) {
+            LOG.debugf("HTTP method '%s' not allowed, this static handler accepts only GET, OPTIONS and HEAD",
+                    context.request().method().name());
+            beforeNextHandler(this.currentClassLoader, context);
+        } else {
+            doHandle(context);
+        }
+
+    }
+
+    private String removeStartSlashes(String value) {
+        return value.replaceAll("^/+", "");
+    }
+
+    private void doHandle(RoutingContext context) {
+        String resolvedPath = resolvePath(context);
+        String path = resolvedPath.endsWith("/") ? resolvedPath.concat(this.indexPage) : resolvedPath;
+
+        if (this.generatedResources.contains(removeStartSlashes(path))) {
+            compressIfNeeded(context, path);
+            byte[] content = getClasspathResourceContent(path);
+            if (content != null) {
```

Comment:
I guess here should be a warning, if the path is found but their is not content no?